tests/io/test_pyarrow.py

```diff
@@ -1718,3 +1720,85 @@ def test_bin_pack_arrow_table(arrow_table_with_null: pa.Table) -> None:
     # and will produce half the number of files if we double the target size
     bin_packed = bin_pack_arrow_table(bigger_arrow_tbl, target_file_size=arrow_table_with_null.nbytes * 2)
     assert len(list(bin_packed)) == 5
+
+
+def test_partition_for_demo() -> None:
+    import pyarrow as pa
```

Comment:
```suggestion

```
Since the test is moved to `test_pyarrow.py`, we do not need to import pyarrow separately in the test. pyarrow is imported for the whole test file.